### DISCUSSION TITLE
Optimization of Gates in GRAPE

### DISCUSSION THREAD

**Discussion by STaherpour** [JuliaQuantumControl/QuantumControl.jl#27] [Q&A] [answered]
Hi 
I want to optimize a Gate in GRAPE, but I couldn't use `Objective` function for this problem. Because it gets just states not logical basis.
Is there the possibility to Optimize Gates in in this package? 
Thank you in advance

**Comment by goerz** [comment, marked as answer]
What do you mean? The package is *designed* to optimize gates. See https://juliaquantumcontrol.github.io/GRAPE.jl/stable/examples/perfect_entanglers/
